### PR TITLE
[fix] Add random delay to app fetchlist cron job

### PR DESCRIPTION
**Pull request by alexAubin**
Fixes https://dev.yunohost.org/issues/892 (All yunohost instances update their appslist at midnight precisely)

### CODE REVIEW

**Comment by M5oul** [comment]
Will the cron job script will be updated for current install?
It seems adding a new script update it.

File: src/yunohost/app.py [hunk 1]
```diff
@@ -1933,8 +1933,14 @@ def _install_appslist_fetch_cron():
...
 
+    cron_job = []
+    cron_job.append("#!/bin/bash")
+    # We add a random delay between 0 and 60 min to avoid every instance fetching
+    # the appslist at the same time every night
+    cron_job.append("(sleep $((RANDOM%3600)); yunohost app fetchlist > /dev/null 2>&1) &")
+
     with open(cron_job_file, "w") as f:
-        f.write('#!/bin/bash\n\nyunohost app fetchlist > /dev/null 2>&1\n')
+        f.write('\n'.join(cron_job))
```
Comment (M5oul) [inline src/yunohost/app.py]: Why joining with a `\n` at file beginning?

**Comment by alexAubin** [comment]
>Will the cron job script will be updated for current install?
>It seems adding a new script update it.

It will because this is included in the dirty migration during the applist management refactoring here

File: src/yunohost/app.py (hunk 1 above)
Comment (alexAubin) [inline src/yunohost/app.py]: Uh, not sure I get the question ? `char.join(list)` concatenate the element of the list with char in between. So if you have `'\n'.join(["abc", "dev"])`, you get `"abc\ndef"`

Comment (M5oul) [inline src/yunohost/app.py]: Ah, ok. I thought it was `\n$cron_job`.

File: src/yunohost/app.py [hunk 2]
```diff
@@ -1933,8 +1933,14 @@ def _install_appslist_fetch_cron():
 
     logger.debug("Installing appslist fetch cron job")
 
+    cron_job = []
+    cron_job.append("#!/bin/bash")
+    # We add a random delay between 0 and 60 min to avoid every instance fetching
+    # the appslist at the same time every night
+    cron_job.append("(sleep $((RANDOM%3600)); yunohost app fetchlist > /dev/null 2>&1) &")
```
Comment (opi) [inline src/yunohost/app.py]: please split this line in 2 separate line, one with the delay, and the other with the `fetchlist` command, to improve readability

**Pull request merged**